test/integration/specialist_document_test.rb

```diff
@@ -2,10 +2,22 @@
 
 class SpecialistDocumentTest < ActionDispatch::IntegrationTest
   test "random but valid items do not error" do
```

Comment:
Minor: these are no longer random, so we could rename this test to: `example items do not error`

Reply:
Shouldn't they still be random?

Reply:
The aim of the random test is to ensure that the app doesn't error on anything regarded as acceptable by the schema.

Reply:
But the `setup_and_visit_random_content_item` doesn't work for specialist document sub types, it only works for top level schemas.  So perhaps we replace this test implementation with just `setup_and_visit_random_content_item(document_type: 'specialist_document')` and keep the random in the name, and add a separate new test with this new implementation and we get both bits of coverage.

Reply:
However, I think the random test is dangerous because if it does fail, chances are someone will simply re-run the test and see it pass and move on with their life, not fixing the randomly generated error.  We'd need the failure to output some very clear instructions on how to re-run it with the same data so that we can debug it and fix it.  Perhaps it could output the content_item it used so we can copy paste it or something?

Reply:
As per @fofr's request, I've left the random examples in as a generic `specialist_document`. If we chose specific examples, these would fail because there aren't any schemas, as Murray explained above. The reason these random example have passed so far was because the `setup_and_visit_random_content_item` method doesn't fail when it can't find the schema you want. It will instead default to fetching a random `specialist_document` example. 

So might as well make it do that in the first place.

Reply:
Also, the `setup_and_visit_random_content_item` was doubly broken, because it was using `schema_type` instead of the parameter `document_type`, which has been fixed here: https://github.com/alphagov/government-frontend/pull/712/files#diff-f14ca9959ff06f3ca2ee6b9564866954L194. 

So it was failing because 
1. mistake in the method which I've fixed now 
2. once it actually attempts to look for any of the original 4 schema types, it won't find them because they are subtypes of `specialist_document` and will never get their own schema. 
Original 4 random schema types used:

```ruby
setup_and_visit_random_content_item(document_type: 'aaib_report')
setup_and_visit_random_content_item(document_type: 'raib_report')
setup_and_visit_random_content_item(document_type: 'tax_tribunal_decision')
setup_and_visit_random_content_item(document_type: 'cma_case')
```

Reply:
any thoughts @fofr ?

Reply:
I think this might have broken in: https://github.com/alphagov/government-frontend/commit/bbc66a44a9b99ebbb251df93d46ccf1ff84455d9#diff-f14ca9959ff06f3ca2ee6b9564866954R194 

It looks like the ability to specify a custom document_type was removed there.

Reply:
Looks like it, I wonder if it's because it would fail occasionally.  The old version would create a random `specialist_document` and then change the `document_type` to the one you supplied.  I think it's very likely that this could produce a document that wasn't valid.  The metadata you need for an `aaib-report` is not the same as the metadata you need for a `utaac-decision` so if `RandomExample` gave you an `aaib-report` and you simply set the `document_type` on it to `utaac-decision`, it would expect it would no longer be valid.

Reply:
Turns out I'm wrong.  The json schema doesn't validate that the `document_type` and `metadata` match up in the `specialist_document` schema.  This feels wrong as a "utaac_decision" with the metadata of an "aaib-report" is not correct even if it is valid, but also, I looked at how to make our json schema do this and it seemed very painful.  I withdraw my objection here.